control-plane-operator/controllers/hostedcontrolplane/oapi/deployment.go

```diff
@@ -240,34 +313,57 @@ func oasContainerMain() *corev1.Container {
 	}
 }
 
-func oasSocks5ProxyContainer() *corev1.Container {
+func oasKonnectivityProxyContainer() *corev1.Container {
 	return &corev1.Container{
-		Name: "socks5-proxy",
+		Name: "konnectivity-proxy",
 	}
 }
 
+const buildTrustAnchorScript = `
+#!/bin/bash
+
+set -euo pipefail
+
+cp -f /etc/pki/ca-trust/extracted/pem/* /run/ca-trust-generated/
+
+if ! [[ -f /run/service-ca-signer/service-ca.crt ]]; then
+   exit 0
+fi
+
+chmod 0666 /run/ca-trust-generated/tls-ca-bundle.pem
+echo '#service signer ca' >> /run/ca-trust-generated/tls-ca-bundle.pem
+cat /run/service-ca-signer/service-ca.crt >>/run/ca-trust-generated/tls-ca-bundle.pem
+chmod 0444 /run/ca-trust-generated/tls-ca-bundle.pem
+`
+
 func buildOASTrustAnchorGenerator(oasImage string) func(*corev1.Container) {
 	return func(c *corev1.Container) {
 		c.Image = oasImage
 		c.Command = []string{
 			"/bin/bash",
 			"-c",
-			"cp -f /etc/pki/ca-trust/extracted/pem/* /run/ca-trust-generated/ && " +
-				"if ! [[ -f /run/service-ca-signer/service-ca.crt ]]; then exit 0; fi && " +
-				"chmod 0666 /run/ca-trust-generated/tls-ca-bundle.pem && " +
-				"echo '#service signer ca' >> /run/ca-trust-generated/tls-ca-bundle.pem && " +
-				"cat /run/service-ca-signer/service-ca.crt >>/run/ca-trust-generated/tls-ca-bundle.pem && " +
-				"chmod 0444 /run/ca-trust-generated/tls-ca-bundle.pem",
+			buildTrustAnchorScript,
 		}
 		c.VolumeMounts = volumeMounts.ContainerMounts(c.Name)
 	}
 }
 
-func buildOASSocks5ProxyContainer(socks5ProxyImage string) func(c *corev1.Container) {
+func buildOASKonnectivityProxyContainer(socks5ProxyImage string, proxyConfig *configv1.ProxySpec, noProxy string) func(c *corev1.Container) {
 	return func(c *corev1.Container) {
 		c.Image = socks5ProxyImage
-		c.Command = []string{"/usr/bin/control-plane-operator", "konnectivity-socks5-proxy"}
+		c.Command = []string{"/usr/bin/control-plane-operator", "konnectivity-https-proxy"}
 		c.Args = []string{"run"}
+		if proxyConfig != nil {
+			if proxyConfig.HTTPProxy != "" {
```

Comment:
do we really need the if checks? i.e. does it have any negative effect if we pass the flags with ""?

Reply:
we don't, will remove